android/src/test/java/com/onegini/mobile/sdk/reactnative/ValidatePinWithPolicyUseCaseTests.kt

```diff
@@ -0,0 +1,79 @@
+
+import com.facebook.react.bridge.Promise
+import com.onegini.mobile.sdk.android.handlers.OneginiPinValidationHandler
+import com.onegini.mobile.sdk.android.handlers.error.OneginiPinValidationError
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onegini.mobile.sdk.reactnative.clean.use_cases.ValidatePinWithPolicyUseCase
+import com.onegini.mobile.sdk.reactnative.exception.OneginiWrapperErrors
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Answers
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.any
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+@RunWith(MockitoJUnitRunner::class)
+class ValidatePinWithPolicyUseCaseTests {
+
+    @get:Rule
+    val reactArgumentsTestRule = ReactArgumentsTestRule()
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    lateinit var oneginiSdk: OneginiSDK
+
+    @Mock
+    lateinit var promiseMock: Promise
+
+    @Mock
+    lateinit var oneginiPinValidationErrorMock: OneginiPinValidationError
+
+    @Test
+    fun `When supplying null as pin should reject with PARAMETERS_NOT_CORRECT error`() {
+        val pin = null
+        Mockito.`when`(oneginiSdk.oneginiClient.userClient.authenticatedUserProfile).thenReturn(UserProfile("testId"))
+        ValidatePinWithPolicyUseCase(oneginiSdk)(pin, promiseMock)
+
+        verify(promiseMock).reject(OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.code, "Expected parameter 'pin' to be String but was NULL")
+    }
+
+    @Test
+    fun `When pin is not null should call validatePinWithPolicy on the onegini sdk`() {
+        val pin = "14789"
+        ValidatePinWithPolicyUseCase(oneginiSdk)(pin, promiseMock)
+
+        verify(oneginiSdk.oneginiClient.userClient, times(1)).validatePinWithPolicy(any(), any())
+    }
+
+    @Test
+    fun `When oginini validatePinWithPolicy calls onSuccess on the handler, promise should resolve with null`() {
+        val pin = "14789"
+        Mockito.`when`(oneginiSdk.oneginiClient.userClient.validatePinWithPolicy(any(), any())).thenAnswer {
+            it.getArgument<OneginiPinValidationHandler>(1).onSuccess()
+        }
+        ValidatePinWithPolicyUseCase(oneginiSdk)(pin, promiseMock)
+
+        verify(promiseMock).resolve(null)
+    }
+
+    @Test
+    fun `When oginini validatePinWithPolicy calls onError on the handler, promise should reject with error from native sdk`() {
+        val pin = "14789"
+        val errorCode = 111
+        val errorMessage = "message"
+
+        Mockito.`when`(oneginiPinValidationErrorMock.errorType).thenReturn(errorCode)
+        Mockito.`when`(oneginiPinValidationErrorMock.message).thenReturn(errorMessage)
+        Mockito.`when`(oneginiSdk.oneginiClient.userClient.validatePinWithPolicy(any(), any())).thenAnswer {
+            it.getArgument<OneginiPinValidationHandler>(1).onError(oneginiPinValidationErrorMock)
+        }
```

Comment:
to improve tests readability in Android native projects we have an approach that if there are more than one `when` call we move them to a separate method. This method could be called something like `whenPinValidationReturnedError()`